File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.swift
import Foundation

func solution(_ s:String) -> Int {
    var temp = ""
    var count1 = 0
    var count2 = 0
    var answer = 0
    for char in s{
        if temp == "" {
            temp = String(char)
        }
        if temp == String(char) {
            count1 += 1
        }
        else {
            count2 += 1
        }
        if count1 == count2 {
            temp = ""
            answer += 1
        }
    }
    return temp == "" ? answer : answer + 1
}

File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.py
def solution(s):
    answer = 0
    temp = ""
    cnt1 = 0
    cnt2 = 0
    for char in s:
        if temp == "":
            temp = char
        if char == temp:
            cnt1 += 1
        else:
            cnt2 += 1
        if cnt1 == cnt2:
            temp = ""
            answer += 1
    if temp != "":
        answer += 1
    return answer